daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -150,20 +143,20 @@ final class ImmArray[+A] private (
 
   /** O(n) */
   def map[B](f: A => B): ImmArray[B] = {
-    val newArray: mutable.ArraySeq[B] = new mutable.ArraySeq(length)
+    val newArray: Array[Any] = new Array(length)
```

Comment:
I’m not super happy about the `Array[Any]` part but that seemed like the most reasonable option to make this work. Open to other suggestions.